### PR TITLE
feat: first time user spotlight animation

### PR DESCRIPTION
**Pull request by nivedin**
Closes HFE-476

### Description
This PR adds animation to the spotlight search component. This will only be visible to first time user who hasn't opened the spotlight yet. Also adds "Import Collection" option in the spotlight.

https://github.com/hoppscotch/hoppscotch/assets/53208152/efbbb320-1d5d-4711-9dcb-a8dd7dcca664

### Checks
- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [x] All the tests have passed

### CODE REVIEW

File: packages/hoppscotch-common/src/components/app/SpotlightSearch.vue [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+import { invokeAction } from "~/helpers/actions"
+import { getPlatformSpecialKey } from "~/helpers/platformutils"
+import { toggleSetting } from "~/newstore/settings"
+
+const t = useI18n()
+
+const HAS_OPENED_SPOTLIGHT = useSetting("HAS_OPENED_SPOTLIGHT")
+
+const words = ref([
+  { text: '"Selecting a Request"', show: true },
```
Comment (JoelJacobStephen) [inline packages/hoppscotch-common/src/components/app/SpotlightSearch.vue:61]: Can we use i18n strings here instead?

**Review by JoelJacobStephen — CHANGES_REQUESTED** [review]
The animation will stop once the user clicks on the spotlight for the first time. However it does not stop if the user uses the shortcut to invoke the spotlight. I think we should also toggle `HAS_OPENED_SPOTLIGHT` when the user uses the shortcut to invoke the spotlight.

File: packages/hoppscotch-common/src/components/app/SpotlightSearch.vue [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        () => {
+          invokeAction('modals.search.toggle', undefined, 'mouseclick')
+          !HAS_OPENED_SPOTLIGHT ? toggleSetting('HAS_OPENED_SPOTLIGHT') : null
+        }
+      "
+    >
+      <span class="inline-flex flex-1 items-center">
+        <icon-lucide-search class="svg-icons mr-2" />
+        <span v-if="!HAS_OPENED_SPOTLIGHT" class="flex flex-1">
+          Try
```
Comment (JoelJacobStephen) [inline packages/hoppscotch-common/src/components/app/SpotlightSearch.vue:22]: We can use i18n strings here as well

File: packages/hoppscotch-common/src/components/app/SpotlightSearch.vue [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+          <TransitionGroup tag="div" name="list" class="ml-1 relative">
+            <span
+              v-for="(word, index) in wordToShow"
```
Comment (JoelJacobStephen) [inline packages/hoppscotch-common/src/components/app/SpotlightSearch.vue:25]: We are showing a phrase/sentence here.  A better naming would be `phrasesToShow` in my opinion.

File: packages/hoppscotch-common/src/components/app/SpotlightSearch.vue [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+const words = ref([
+  { text: '"Selecting a Request"', show: true },
+  { text: '"Switching Environment"', show: false },
+  { text: '"Changing Theme"', show: false },
+])
+
+let intervalId: ReturnType<typeof setTimeout> | null = null
+
+//cycle through the words
+const showNextWords = () => {
```
Comment (JoelJacobStephen) [inline packages/hoppscotch-common/src/components/app/SpotlightSearch.vue:69]: A better name would be `showNextPhrase`

File: packages/hoppscotch-common/src/components/app/SpotlightSearch.vue (hunk 1 above)
Comment (nivedin) [reply packages/hoppscotch-common/src/components/app/SpotlightSearch.vue]: yes

**Review by JoelJacobStephen — CHANGES_REQUESTED** [review]
Feature wise, everything is working as expected. I have suggested minor code improvements. Once they are resolved, it's good to go from my end.

File: packages/hoppscotch-common/src/services/spotlight/searchers/collections.searcher.ts [hunk 5]
```diff
@@ -183,41 +193,45 @@ export class CollectionsSpotlightSearcherService
...
-                    componentProps: {
-                      folderPath: result.id.split("rest-")[1],
-                    },
-                  },
-            icon: markRaw(IconFolder),
+            text: getResultText(result.id),
+            icon: getResultIcon(result.id),
             score: result.score,
           }))
         } else if (pageCategory === "graphql") {
```
[the remark this answers is not in the document]
Comment (nivedin) [reply packages/hoppscotch-common/src/services/spotlight/searchers/collections.searcher.ts]: yeah good catch.. patched

**Review by JoelJacobStephen — APPROVED** [review]
LGTM!! 💯

**Review by AndrewBastin — APPROVED** [review]

**Comment by chlab** [comment]
I was about to open an issue complaining about this animation as it's kind of distracting when keeping hoppscotch open on a secondary display and something keeps trying to get your attention :). While searching for existing issues I found this and realized I can just use the search once to turn it off.

Just wanted to drop in and say thank you for being considerate/mindful enough to turn it off once the search was used! ❤️  And while I'm at it, thanks to all HS contributors for a great API client!

**Pull request merged**